repository/service.proto

```diff
@@ -224,6 +229,9 @@ message EventlogItem {
   string old_language   = 12;
   // MainDocument UUID if the document is a metadata document.
   string main_document  = 13;
+  // SystemState is set to "restoring" if this event was generated from a
+  // restore operation.
+  string system_state   = 14;
```

Comment:
Yeah, an enum is not a bad idea. The default system state is `""`, i.e. no system state.

Reply:
Rediscovered that enums are a bit annoying, we gain that the return values are properly specified, but lose the ability to let empty/zero value communicate no state:

```proto
enum SystemState {
  SYSTEM_STATE_UNKNOWN   = 0;
  SYSTEM_STATE_NONE      = 1;
  SYSTEM_STATE_RESTORING = 2;
}
```

Best practice in protobuf is to let 0 represent an unknown value (f.ex. if a client or server run off different specs and sends a hypothetical 3 that value will be recieved as 0). That has the best forward compatability as it's better if a client is informed that "this is a state you don't know about" rather than "this has no state". So that would be 1 for NONE and 2 for RESTORING:

```diff
@@ -165,7 +175,7 @@
     "updater_uri": "user://test/testdeleterestore",
     "type": "core/article",
     "language": "en-gb",
-    "system_state": "restoring"
+    "system_state": 2
   },
   {
     "id": 15,
@@ -184,7 +194,7 @@
     "updater_uri": "user://test/testdeleterestore",
     "type": "core/article",
     "language": "en-gb",
-    "system_state": "restoring"
+    "system_state": 2
   },
   {
     "id": 16,
@@ -194,6 +204,7 @@
     "version": 2,
     "updater_uri": "user://test/testdeleterestore",
     "type": "core/article",
-    "language": "en-gb"
+    "language": "en-gb",
+    "system_state": 1
   }
 ]
```

When communicating with the JSON API the numbers are encoded as  "SYSTEM_STATE_NONE|UNKNOWN|RESTORING". I don't know what you think but I'm a bit meh about switching to enums now :smile:

Reply:
There's really no enum support in either JSON or JavaScript, but TypeScript does support defining types as sets of strings, so it would probably work fine on that end.

That said, I don't feel this is absolutely necessary either. Just documenting the values in the comment string could be enough.